### DISCUSSION TITLE
Needs tips on coordinating components on page (using $store for example)

### DISCUSSION THREAD

**Discussion by cjaoude** [livewire/livewire#3884] [1. Help]
When a page loads, with many components, one challenge is coordinating data between the many components. One way is to fire off an event with the data payload, some issues:

- on page load, firing an event to set a default value, some components won't hear the event due to the order/priority of the components loading on the page. Basically, it can fire before they are ready. Any tips here?

- firing events among many components, it gets messy fast. So I attempted to use Alpine.store and @entangle. It actually works, but for unknown reasons (and debugging fails here), some components fail to read the store even though the have the same bit of code for reading the $store as components where it works fine.

Overall, I'd appreciate any strategies for having many components on the page and how to coordinate values reactively among them. Having the $store (or other solution) work properly if used like this.

I'll share this a working example of how I've managed to share the store with the LW Component

```
// each component would need it's own version of this, 
// but the `sharedDate` from `$store` would globally update components 
// (except some components don't for reasons unknown

<script>
    document.addEventListener('alpine:init', () => {
        Alpine.data('dailyData', () => ({
            '$store.datePicker.sharedDate': @entangle('dateSelected'),
        }));
    });
</script>
```

**Comment by joshhanley** [comment]
@cjaoude honestly, at the moment, there isn't a great strategy. Personally I recommend not splitting into multiple Livewire components, instead I use one main Livewire component for a page and make heavy use of blade components. I only create nested child components if I really have to and if there isn't a lot of interaction between components.

I think the reason `$store` and `@entangle` don't work for all components, is each time it gets assigned only the proxy for the last component assigned will be active, as such that will be the only one that works.

If you need to communicate between components, I'd probably stick to events for now, but it is something we are looking at but it's not a simple change.

Hope this helps!
